### PR TITLE
feat: remove beta field validations

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR removes all traces of `beta-permissions` and removes the need to validate each field's creatability based on beta flags, since beta flags have been removed.

Part of #8

## Solution

**Features**:

- remove `utils/beta-permissions`

**Improvements**:
- Remove requirement for base field schema to be created as a function, since there is no more need to check the User collection when creating a form field.

### CODE REVIEW

File: src/app/utils/beta-permissions.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { BasicField, IFieldSchema, IUserSchema } from '../../types'
+
+/**
+ * Array contains all current beta fields. Add or remove field types as the beta
+ * fields change.
+ */
+const BETA_FIELDS = [BasicField.Mobile]
```
Comment (mantariksh) [inline src/app/utils/beta-permissions.ts:7]: I think this was my mistake while removing `allowSms`, mobile is no longer a beta field so this array should be empty

**Review by mantariksh — CHANGES_REQUESTED** [review]

File: src/app/models/field/baseField.ts [hunk 2]
```diff
@@ -99,7 +96,7 @@ const createBaseFieldSchema = (db: Mongoose) => {
     // Check if user is allowed to add this field.
     if (isBetaField(thisField)) {
       User.findById(admin, function (err, user) {
-        if (err) {
+        if (err || !user) {
```
[the remark this answers is not in the document]
Comment (karrui) [reply src/app/models/field/baseField.ts:99]: You are right. Fixed the control flow.

Will probably need tests too -- issue has been created and tracked at https://github.com/opengovsg/FormSG/issues/102

**Review by mantariksh — APPROVED** [review]

File: src/app/models/field/baseField.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (liangyuanruo) [reply src/app/models/field/baseField.ts:99]: should we move `return next()` into the callback then?

[the remark this answers is not in the document]
Comment (karrui) [reply src/app/models/field/baseField.ts]: Not a problem anymore, removed the checks here since we have no more beta flags.

**Comment by karrui** [comment]
Require a re-review, since the scope has changed since the beginning of this PR. Removed all traces of beta-permissions instead since we do not have anymore beta flags.

**Review by mantariksh — APPROVED** [review]

**Pull request merged**